c10/util/BFloat16.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+// Defines the bloat16 type (brain floating-point). This representation uses
+// 1 bit for the sign, 8 bits for the exponent and 7 bits for the mantissa.
+
+#include <c10/macros/Macros.h>
+#include <iostream>
+#include <math.h>
+#include <cmath>
+#include <stdint.h>
+#include <inttypes.h>
+#include <stdio.h>
+
+namespace c10 {
+
+namespace detail {
+  inline float f32_from_bf16(uint16_t src) {
```

Comment:
this is strange because its `from_bf16`, but this doesn't take a BFloat16.  The half equivalent is `from_bits` which makes more sense.